### ISSUE TITLE
.tar from .obscpio has to be compressed, otherwise validator fails

### ISSUE THREAD

**Issue by jirislaby** [openSUSE/obs-service-source_validator#84]
See:
https://build.opensuse.org/request/show/823140

I create a .tar from .obscpio there. And source_validator fails to pass.

If I create .tar.xz as in https://build.opensuse.org/request/show/823294, it suddenly succeeds.

**Comment by adrianschroeter** [comment]
you need indeed a buildtime compression after the "tar" service when using obs_scm

 <service name="recompress" mode="buildtime">

The reason is that otherwise the uncompressed tar ends up in the src.rpm with no or poor compression.

**Comment by jirislaby** [comment]
But https://github.com/openSUSE/obs-service-tar_scm says:

> Having said that, it may be more efficient to drop the build-time usage of recompress, since at build-time rpmbuild would decompress the same file soon after compressing it. In this case, only the tar source service would be used to reconstruct an uncompressed tarball to be consumed by the .spec file. However this has the side-effect that the resulting .src.rpm will contain an uncompressed tarball too. This is not necessarily a problem because .src.rpm files are compressed anyway, and in fact it may even be more efficient to avoid double compression (i.e. the sources within the .src.rpm, and the .src.rpm itself both being compressed). But this depends very much on the combination of compression formats used for compression of the sources, and for compression of the .src.rpm. Therefore the decision whether to use recompress will depend on what format is desired within the resulting .src.rpm, and on the types of compression being used for both the tarball and by rpmbuild for constructing the source rpms.

**Comment by adrianschroeter** [comment]
yes, that is true in general, but openSUSE distro has stricter policy.

obs_scm is a generic service, but source_validator applies the SUSE policies which are stricter.

And neither factory nor leap seems to have source compression enabled atm.

**Comment by jirislaby** [comment]
OK, the error message could be more helpful then :). As I tried running `osc service localrun source_validator` locally without any output.

**Comment by coolo** [comment]
The problem is that source_validator disables all checks if there is a _service file. And that error message comes from times before the existance of buildtime services.

**Comment by coolo** [comment]
But I actually don't see any 'SUSE policy' stricking here. This seems to be a rather strange bug I will continue looking into

**Comment by coolo** [comment]
```if test -f "$DESTINATIONDIR/${file/\.tar\.*/}.obscpio"; then``` is the problem. If it does stop expecting something behind .tar the bug is gone

**Comment by adrianschroeter** [comment]
On Mittwoch, 29. Juli 2020, 09:00:01 CEST wrote Stephan Kulow:
> But I actually don't see any 'SUSE policy' stricking here. This seems to be a rather strange bug I will continue looking into

I am sure that we do not want to ship uncompressed tar balls in src.rpm's ...

-- 

Adrian Schroeter
email: adrian@suse.de

SUSE Linux GmbH, GF: Felix Imendörffer, Jane Smithard, Graham Norton, HRB 21284 (AG Nürnberg) 
 
Maxfeldstraße 5                         
90409 Nürnberg 
Germany

**Comment by coolo** [comment]
Possibly, but that's not for 20-files-and-present to check :)

**Comment by coolo** [comment]
src.rpms have gzip compression, so they are not as efficient as xz but also way less time comsuming.

-rw-r--r-- 1 coolo  399 55759 29. Jul 09:17 compressed.src.rpm
-rw-r--r-- 1 coolo suse 62670 29. Jul 09:16 uncompressed.src.rpm
while b4-0.5.2.tar is 225280

**Comment by coolo** [comment]
So please don't use 'SUSE policy' as excuse :)

**Comment by jirislaby** [comment]
> -rw-r--r-- 1 coolo 399 55759 29. Jul 09:17 compressed.src.rpm
> -rw-r--r-- 1 coolo suse 62670 29. Jul 09:16 uncompressed.src.rpm

Isn't it xz vs. gz instead?

Anyway, I sort of expected "some kind of compression" in src.rpm. And in this particular package, I didn't bother to compress the tar provided the size as the difference between "some kind of compression" and xz would be neligible.

**Comment by adrianschroeter** [comment]
On Mittwoch, 29. Juli 2020, 09:20:22 CEST wrote Stephan Kulow:
> So please don't use 'SUSE policy' as excuse :)

okay, but please do not "fix" it without adding a new check
resulting in same behaviour ;)

-- 

Adrian Schroeter
email: adrian@suse.de

SUSE Linux GmbH, GF: Felix Imendörffer, Jane Smithard, Graham Norton, HRB 21284 (AG Nürnberg) 
 
Maxfeldstraße 5                         
90409 Nürnberg 
Germany

**Comment by coolo** [comment]
> Isn't it xz vs. gz instead?
>
It's tar.xz.gz vs tar.gz in essense

**Comment by coolo** [comment]
Again, what is the policy you're expecting? we're far behind the times when we recompressed sources to .bz2 so they fit on a DVD. So the gz compression from rpm should be fine

**Comment by adrianschroeter** [comment]
On Mittwoch, 29. Juli 2020, 09:28:25 CEST wrote Stephan Kulow:
> 
> > Isn't it xz vs. gz instead?
> >
> It's tar.xz.gz vs tar.gz in essense

btw, we have this line in rpm config:

#%_source_payload       w9.gzdio

so it looked like disabled compression, but macros 
get expanded independend of bash syntax ...

So, it should be decided to stick uncompressed
as intendended or with gz compression as configured ;)

but still, if we change this for tumbleweed we still
need to consider to be backward compatible with the
check in source_validator.

-- 

Adrian Schroeter
email: adrian@suse.de

SUSE Linux GmbH, GF: Felix Imendörffer, Jane Smithard, Graham Norton, HRB 21284 (AG Nürnberg) 
 
Maxfeldstraße 5                         
90409 Nürnberg 
Germany

**Comment by adrianschroeter** [comment]
On Mittwoch, 29. Juli 2020, 09:29:23 CEST wrote Stephan Kulow:
> Again, what is the policy you're expecting? we're far behind the times when we recompressed sources to .bz2 so they fit on a DVD. So the gz compression from rpm should be fine

well, for the future I would even consider to drop source rpms
complete and provide the sources independend via OBS 
mechanics. 

This avoids wasted space, because we can de-duplicate way better
and make pieces of the sources easier accessable.

But this is a bigger task and needs further discussions.

-- 

Adrian Schroeter
email: adrian@suse.de

SUSE Linux GmbH, GF: Felix Imendörffer, Jane Smithard, Graham Norton, HRB 21284 (AG Nürnberg) 
 
Maxfeldstraße 5                         
90409 Nürnberg 
Germany

**Comment by coolo** [comment]
In any case: SUSE policies belong in rpmlint - the job of source validator is to check if the sources are valid, not if they match the policy
